### ISSUE TITLE
Banned card added to generated deck: standard - Emrakul, the Promised End

### ISSUE THREAD

**Issue by JayDi85** [magefree/mage#4056]
Steps to reproduce:
1. Open generate window;
2. Choose Standard;
3. Select green colour;
4. Selecte deck size 60;
5. Press OK.

Sometime it add "Emrakul, the Promised End", but it banned from standard and that generated deck can't play (see screenshot).

![shot_170925_031807](https://user-images.githubusercontent.com/8344157/30788158-a2ef9d42-a1a8-11e7-844f-4e089013d1b3.png)

**Comment by Simown** [comment]
I'll have a look at this when I get a minute, but probably won't be until next week.

**Comment by theelk801** [comment]
I looked at it and forgot to comment. It's definitely confirmed, but fixing the issue seems a bit complicated. I can also confirm that it applies to other formats as well.

**Comment by Simown** [comment]
Oh, that's good to know you looked at it. I reworked the generator a while ago so thought I'd have a good grasp of what's happening. 

Looking at it earlier, it seems the cause is that the generator seems to only check the sets to generate from without filtering out the current banlist.

**Comment by theelk801** [comment]
Yeah, I tried to see if I could implement a fix but it got hairy pretty quickly and I figured that just generating a new deck until you got a legal one would be an acceptable workaround in the meantime.

**Comment by Simown** [comment]
Yeah, by no means a huge issue as you don't even know what's in your deck anyway and can be fixed usually by just pressing the button again. Would be nice to fix either by either manually filtering the banlist or making the database never returning the cards in the first place that aren't legal.

Thanks for looking at it already. I'll see if I have any more luck!

**Comment by JayDi85** [comment]
Can't be fixed cause dialog don't have deck type information. Current deck editor has built in deck validator, so that's problem is outdated -- just generate new deck or fix existing on invalid.
